community/kernel/src/test/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStoreTest.java

```diff
@@ -0,0 +1,412 @@
+                        {
+                            @Override
+                            public void write( WritableBuffer key, WritableBuffer value )
+                            {
+                                key.putByte( 0, (byte) 'f' );
+                                key.putByte( 1, (byte) 'o' );
+                                key.putByte( 2, (byte) 'o' );
+                                value.putInt( 0, data );
+                            }
+                        }
+                ) );
+                old.close();
+                files[i] = file.first();
+            }
+            file.other().close();
+        }
+        // Corrupt the last files
+        try ( StoreChannel channel = the.fileSystem().open( files[9], "rw" ) )
+        {   // ruin the header
+            channel.position( 16 );
+            ByteBuffer value = ByteBuffer.allocate( 16 );
+            value.put( (byte) 0 );
+            value.flip();
+            channel.writeAll( value );
+        }
+        try ( StoreChannel channel = the.fileSystem().open( files[8], "rw" ) )
+        {   // ruin the header
+            channel.position( 32 );
+            ByteBuffer value = ByteBuffer.allocate( 16 );
+            value.put( (byte) 17 );
+            value.flip();
+            channel.writeAll( value );
+        }
+        try ( StoreChannel channel = the.fileSystem().open( files[7], "rw" ) )
+        {   // ruin the header
+            channel.position( 32 + 32 + 32 + 16 );
+            ByteBuffer value = ByteBuffer.allocate( 16 );
+            value.putLong( 0 );
+            value.putLong( 0 );
+            value.flip();
+            channel.writeAll( value );
+        }
+
+        // then
+        try ( Lifespan life = new Lifespan() )
+        {
+            life.add( store );
+
+            assertEquals( 64, lastLong( store.metadata().get( "txId" ) ) );
+        }
+    }
+
+    static DataProvider data( final Entry... data )
+    {
+        return new DataProvider()
+        {
+            int i;
+
+            @Override
+            public boolean visit( WritableBuffer key, WritableBuffer value ) throws IOException
+            {
+                if ( i < data.length )
+                {
+                    data[i++].write( key, value );
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            public void close() throws IOException
+            {
+            }
+        };
+    }
+
+    interface Entry
+    {
+        void write( WritableBuffer key, WritableBuffer value );
+    }
+
+    static byte[] txId( long txId )
+    {
+        return new BigEndianByteArrayBuffer( 16 ).putLong( 8, txId ).buffer;
+    }
+
+    static long lastLong( byte[] txIds )
+    {
+        BigEndianByteArrayBuffer lhsTxId = new BigEndianByteArrayBuffer( txIds );
+        return lhsTxId.getLong( lhsTxId.size() - 8 );
+    }
+
+    @Rotation(Rotation.Strategy.INCREMENTING)
```

Comment:
Yes, but quite a few of the tests in here depend on that particular strategy. Although it might make sense for another test to run parameterised with different rotation strategies.